build.gradle

```diff
@@ -128,14 +146,49 @@ subprojects {
     }
 
     compileJava.dependsOn saveProtoNames
+
+    publishing {
+        publications {
+            mavenJava(MavenPublication) {
+                groupId "${group}"
+                artifactId "${project.name}"
+                version "${project.version}"
+
+                from components.java
+            }
+        }
+    }
+    publishing {
+        repositories {
+            maven {
+                credentials {
+//                    username project.extensions.extraProperties.properties.get("repositoryUserName")
+//                    password project.extensions.extraProperties.properties.get("repositoryUserPassword")
+                    username "${repositoryUserName}"
+                    password "${repositoryUserPassword}"
+                }
+                url "${project.MAVEN_REPOSITORY_URL}"
+            }
+        }
+    }
 }
 
-task removeProtoNames << {
-    File propsFile = file(getProtoPropertiesFilePath())
-    if (propsFile.exists()) {
-        propsFile.delete();
+task readPublishingCredentials << {
+    if (repositoryUserName == null || repositoryUserPassword == null) {
+        throw new InvalidUserDataException("Please set up valid credentials.")
```

Comment:
Please add description with the name of the file and its placement into the exception message text.